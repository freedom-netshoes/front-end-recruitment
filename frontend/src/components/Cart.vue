<template>
  <div>
    <transition name="fade" appear mode="out-in">
      <div class="overlay" v-if="cart.show"></div>
    </transition>
    <div class="cart" v-bind:class="{ opened: cart.show }">

      <div class="btn-action" @click="actionCart()">
        <span v-show="cart.show">&#8594;</span>
        <span v-show="!cart.show">&#x2190;</span>
      </div>

      <div class="content-cart">
        <div class="title">
          <div class="bag">
            <div class="image">
              <img src="../../static/img/cart/icon-bag.png" height="40" width="33" alt="">
              <div class="badget">{{cart.products.length}}</div>
              <span>SACOLA</span>
            </div>
          </div>
        </div>

        <ul class="cart-products">
          <transition-group name="fade" appear mode="out-in">
            <li v-for="(product, index) in cart.products" :key="index">
              <div class="image"><img :src="'../../static/img/store/' + product.title.toLowerCase().replace(/ /g, '-') + '_thumb.jpg'" alt="placeholder+image"></div>
              <div class="description">
                <h3>{{product.title}}</h3>
                <p>{{product.size}} | {{product.style}}</p>
                <p>Quantidade: {{product.quantify}}</p>
              </div>
              <div class="price">
                <div class="btn-remove" @click="removeProduct(product, index)">X</div>
                <div class="price-total">{{product.currencyFormat}} {{product.price.toFixed(2).toString().replace('.',',')}}</div>
              </div>
            </li>
          </transition-group>
        </ul>

        <div class="total">
          <div class="sub-total">
            <p>SUBTOTAL</p>
          </div>

          <div class="parcels">
            <h3>R$ {{cart.total.toFixed(2).toString().replace('.',',')}}</h3>
            <p>ou em at√© 10 X R$ {{(cart.total / 10).toFixed(2).toString().replace('.',',')}}</p>
          </div>
        </div>

        <div class="btn-buy">
          <button type="button" @click="buy" :disabled="cart.products.length == 0" :class="[cart.products.length == 0 ? 'disabled' : '']">COMPRAR</button>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'Cart',
  methods: {
    buy () {
      this.$store.commit('BUY', {
        show: false
      })

      this.$store.commit('SNACK_BAR', {
        message: 'Compra realizada com sucesso!',
        timer: 2000,
        status: 'success'
      })
    },
    actionCart () {
      this.$store.commit('SHOW_AND_HIDE_CART', !this.$store.state.cart.show)
    },
    removeProduct (product, index) {
      this.$store.commit('REMOVE_PROD_CART', {
        index: index,
        product: product
      })
    }
  },
  computed: {
    cart () {
      return this.$store.state.cart
    }
  },
  data () {
    return {

    }
  }
}
</script>

<style lang="scss" scoped>
  .overlay {
    position: fixed;
    width: 100%;
    height: 100%;
    background: none rgba(255,255,255, 0.8);
    top: 0;
  }

  .cart {
    position: fixed;
    top: 0;
    right: -45%;
    width: 45%;
    height: 100%;
    background: #202025;
    z-index: 2;
    transition: right .5s ease;

    .content-cart {
      height: 100%;
      overflow-y: scroll;
      overflow-x: hidden;
    }

    .btn-action {
      position: absolute;
      left: -7%;
      top: 0;
      background: #202025;
      color: white;
      font-weight: bold;
      width: 7%;
      padding: 3% 0;
      /* height: 20%; */
      text-align: center;
      border-radius: 10px 0px 0px 10px;
      cursor: pointer;

      &:hover {
        box-shadow: 0 0 45px black inset;
      }
    }

    .title {
      padding: 5% 0;
      width: 100%;

      .bag {
        margin: 1% auto;

        .badget {
          position: absolute;
          left: 3%;
          bottom: -15%;
          background: #dfbd00;
          border-radius: 50%;
          padding: 1%;
          font-size: .9vw;
          line-height: .8;
          font-weight: bold;
        }

        .image {
          position: relative;
          left: 45%;
        }

        span {
          color: white;
          padding: 0;
          position: relative;
          top: -10px;
          left: 1%;
          font-weight: bold;
        }
      }
    }

    ul.cart-products {
      width: 100%;
      margin: 0 auto;

      li {
        width: 100%;
        float: left;
        color: white;
        border-top: 1px solid black;
        padding: 5% 0%;
        margin: 0% 0;

        .image, .description, .price {
          float: left;
        }

        .image {
          width: 15%;
          padding-left: 3%;

          img {
            width: 100%;
          }
        }

        .description {
          width: 50%;
          margin-left: 3%;

          h3 {
            margin-bottom: 1%;
          }

          p {
            margin: 2% 0 0;
            font-size: 1vw;
            color: #999999;
          }
        }

        .price {
          float: right;
          padding-right: 3%;

          .btn-remove {
            cursor: pointer;
            color: black;
            font-weight: bold;
            font-size: 1.5vw;
            text-align: right;
            width: 100%;
            margin-bottom: 25%;

            &:hover {
              color: red;
            }
          }

          .price-total {
            color: #dfbd00;
            font-weight: bold;
          }
        }
      }

      li.removed {
        background: #131316;
        opacity: .5;

        .description {
          text-decoration: line-through;
        }
        .price-total {
          text-decoration: line-through;
        }
      }

      li:last-child {
        border-bottom: 1px solid black;
      }
    }

    .total {
      float: left;
      width: 95%;
      margin-top: 5%;
      padding-left: 3%;

      .sub-total {
        float: left;
        color: #949495;
        margin-top: 2%;
      }

      .parcels {
        float: right;
        text-align: right;

        h3 {
          font-weight: bold;
          font-size: 2vw;
          color: #dfbd00;
          margin-bottom: 5%;
        }

        p {
          color: #5a5a5c;
        }
      }
    }

    .btn-buy {
      float: left;
      width: 95%;
      padding-left: 2.5%;
      margin-bottom: 5%;

      button {
        background: black;
        color: #fff;
        text-align: center;
        width: 100%;
        border: 0;
        height: 50px;
        margin-top: 5%;
        cursor: pointer;

        &:hover {
          background: green;
        }
      }

      button.disabled {
        cursor: not-allowed;
        color: #4e4e4e;
        background: black;

        &:hover {
          background: black;
          color: #4e4e4e;
        }
      }
    }
  }

  .cart.opened {
    right: 0;
  }
</style>
