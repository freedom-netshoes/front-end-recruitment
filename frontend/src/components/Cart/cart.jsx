import React, { Component } from 'react'
import { connect } from 'react-redux'
import { ReduceList } from '../../resources/helpers'

import CSSModules from 'react-css-modules'
import Header from './Header/header'
import BagItem from './BagItem/bag-item'
import SubTotal from './Subtotal/sub-total'
import styles from './cart.less'

@CSSModules(styles)
class Cart extends Component {
  render () {
    return (
      <aside id='cart-shop' className={this.props.cartState ? 'open' : 'close'}>
        <div className='col-xs-12'>
          <Header {...{
            size: this.props.fullBag.length,
            onClick: this.closeBag
          }} />
          <ul>
            {
              this.props.bagItems.map(product =>
                <BagItem {...{
                  product: product,
                  quantity: product.quantity,
                  key: product.id
                }} />
              )
            }
            <hr />
            <li className={this.props.fullBag.length ? 'show' : 'hide'} >
              <SubTotal {...{
                maxInstallment: this.props.maxInstallment,
                priceTotal: this.props.priceTotal
              }} />
            </li>
            <li className={!this.props.fullBag.length ? 'show' : 'hide'} >
              <p className='col-xs-12' styleName='subtotal'>Seu carrinho est√° vazio.</p>
            </li>
          </ul>
        </div>
      </aside>
    )
  }
}
const mapStateToProps = state => {
  const bagItems = ReduceList(state.shopCart)
  const total = state.shopCart.reduce((a, b) => {
    return a + b.price
  }, 0)
  const maxInstallment = Math.max(...bagItems.map(v => v.installments))
  return {
    maxInstallment: maxInstallment,
    priceTotal: total,
    fullBag: state.shopCart,
    bagItems: ReduceList(state.shopCart),
    cartState: state.cartState
  }
}
export default connect(mapStateToProps)(Cart)
