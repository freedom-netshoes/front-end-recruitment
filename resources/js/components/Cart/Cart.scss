@import '../../../scss/shared';

.form {
  background-color: $dark-gray;
  height          : 100%;
}

.cart {
  @include transition();
  @include user-select(none);
  height  : 100%;
  overflow: hidden;
  right   : 0;
  position: absolute;
  top     : 0;
  width   : 1px;
  //min-width   : 66.66%;

  li {
    padding-bottom: 1rem;
    text-align    : left;

    &.cart-row {
      @include transition();
      cursor: pointer;

      &:hover {
        background-color: $darker-gray;
      }
    }

    hr {
      margin          : 0 1rem 1rem;
      background-color: $darker-gray;
      border          : none;
      height          : 2px;
      width           : 100%;
    }
  }
}

:global {
  .open {
    box-shadow: -2px 0px 1px $light-gray;
    width     : 66.66%;
  }

  .delete {
    background-color: $darker-gray;
  }
}
