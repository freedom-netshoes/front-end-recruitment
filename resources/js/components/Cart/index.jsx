import React, {Component, PropTypes} from 'react';
import CartTitle from 'CartTitle';
import Delete from 'Delete';
import Installments from 'Installments';
import CSSModules from 'react-css-modules';
import Quantity from 'Quantity';
import Photo from 'Photo';
import Price from 'Price';
import Size from 'Size';
import Submit from 'Submit';
import Subtotal from 'Subtotal';
import Title from 'Title';
import styles from './Cart.scss'

@CSSModules(styles)
class Cart extends Component {

  static defaultProps = {

  };

  static propTypes = {

  };

  render () {
    return (
      <aside {...{
        className: 'open',
        id: 'cart',
        styleName: 'cart'
      }}>
        <form className="col-xs-12" styleName='form' role="form">
          <CartTitle />
          <ul>
            <li className='row middle-xs delete' styleName='cart-row'>
              <hr />
              <div className='col-xs-12 col-sm-9 start-sm'>
                <Photo />
                <Title />
                <Size />
                <Quantity />
              </div>
              <div className='col-xs-12 col-sm-3 end-xs'>
                <Delete />
                <Price />
              </div>
            </li>
            <li className='row middle-xs' styleName='cart-row'>
              <hr />
              <div className='col-xs-12 col-sm-9 start-sm'>
                <Photo />
                <Title />
                <Size />
                <Quantity />
              </div>
              <div className='col-xs-12 col-sm-3 end-xs'>
                <Delete />
                <Price />
              </div>
            </li>
            <li className='row conclude'>
              <hr />
              <Subtotal />
              <div className='col-xs-12 col-sm-6 end-sm'>
                <Price />
                <Installments />
              </div>
              <Submit />
            </li>
          </ul>
        </form>
      </aside>
    );
  }
}

export default Cart;
