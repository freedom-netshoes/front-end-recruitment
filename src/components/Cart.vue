<template>
  <div
    class="cart"
    :style="style"
   >
   <div
      class="close"
      @click="$emit('close')"
    >
    <CloseIcon class="close-icon"/>
  </div>
    <div class="title">
      <div class="bag">
        <BagIcon class="bag-icon"/>
        <div class="counter">
          {{ $store.getters.cart.length }}
        </div>
      </div>
      <div class="content">
        sacola
      </div>
    </div>
    <div class="list">
      <div
        v-for="item in $store.getters.cart"
        class="product"
        :key="item.id"
      >
        <div class="image">
          <img :src="getImagePath(item)"/>
        </div>
        <div class="info-container">
          <div class="info">
            <div class="title">
              {{ item.title }}
            </div>
            <div class="size-n-style">
              <div class="size">
                {{ formatSizes(item.availableSizes) }}
              </div>
              <div
                v-if="item.availableSizes.length > 0 && item.style.length > 0"
                class="separator"
              />
              <div class="style">
                {{ item.style }}
              </div>
            </div>
            <div class="amount">
              Quantidade: 2
            </div>
          </div>
          <div class="info-extra">
            <div
              class="close"
              @click="removeItem(item)"
              @mouseover="deleteIntention"
              @mouseleave="noDeleteIntention"
            >
              <CloseIcon class="close-icon"/>
            </div>
            <div class="amount">
              <div class="currency">
                R$
              </div>
              <div class="price">
                {{ getMainPrice(item.price) }}
              </div>
              <div class="price-decimal">
                {{ getDecimalPrice(item.price) }}
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="subtotal">
      subtotal
      <div class="amount-container">
        <div class="amount">
          <div class="currency">
            R$
          </div>
          <div class="price">
            {{ getFinalPrice() }}
          </div>
          <div class="price-decimal">
            {{ getFinalDecimalPrice() }}
          </div>
        </div>
        <div class="installments">
          {{ getFinalInstallmentsMessage() }}
        </div>
      </div>
    </div>
    <button class="button big">
      comprar
    </button>
  </div>
</template>
<script>
import { getMainPrice, getDecimalPrice, getCartFinalPrice, getProInstallmentPrice } from '../util/product'
import { REMOVE_FROM_CART } from '../store'
import CloseIcon from './icons/CloseIcon'
import BagIcon from './icons/BagIcon'

export default {
  name: 'Cart',
  components: {
    CloseIcon,
    BagIcon
  },
  props: {
    open: {
      type: Boolean,
      required: true
    }
  },
  computed: {
    style: function () {
      return {
        right: this.open ? '0' : '-600px'
      }
    },
    finalPrice: function () {
      return getCartFinalPrice(this.$store.getters.cart)
    }
  },
  methods: {
    getImagePath: function (item) {
      return require('../assets/' + item.imagePath)
    },
    deleteIntention: function (event) {
      const productEl = this.getParentProductEl(event.target)
      productEl.classList.add('delete-intention')
    },
    noDeleteIntention: function (event) {
      const productEl = this.getParentProductEl(event.target)
      productEl.classList.remove('delete-intention')
    },
    getParentProductEl: function (el) {
      return el.closest('.product')
    },
    removeItem: function (item) {
      this.$store.commit(REMOVE_FROM_CART, item)
      this.$notify({ title: 'Removido do Carrinho' })
    },
    formatSizes: function (sizes) {
      return sizes.map(size => size).join(' ')
    },
    getMainPrice: function (price) {
      return getMainPrice(price)
    },
    getDecimalPrice: function (price) {
      return getDecimalPrice(price)
    },
    getFinalPrice: function () {
      return getMainPrice(this.finalPrice)
    },
    getFinalDecimalPrice: function () {
      return getDecimalPrice(this.finalPrice)
    },
    getFinalInstallmentsMessage: function (item) {
      return `ou em at√© 10 x R$ ${getProInstallmentPrice(this.finalPrice, 10)}`
    }
  }
}
</script>
<style scoped lang="scss">
  .list {
    margin-bottom: 20px;
    .product:last-child {
      border-bottom: 2px solid black;
    }
  }

  .button {
    margin-top: 20px;
  }

  .amount {
    color: $yellow;
    display: flex;
    .currency {
      margin-right: $spacing-xsm;
    }
    .price {
      font-weight: bold;
    }
  }

  .info-container {
    display: flex;
    flex-grow: 1;
    justify-content: space-between;
  }

  .info-extra {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    padding: 5px 0;
  }

  .info {
    display: flex;
    flex-direction: column;
    .title {
      color: white;
    }

    .size-n-style {
      display: flex;
      align-items: center;
      .separator {
        width: 1px;
        height: 1em;
        margin: 0 $spacing-xsm;
        background: $grey-second;
      }
    }
    .size,
    .amount,
    .size-n-style {
      color: $grey-second;
    }
  }

  .subtotal {
    display: flex;
    text-transform: uppercase;
    justify-content: space-between;
    align-items: center;
    padding: 0 10px;
    color: $grey;
    .amount-container {
      display: flex;
      flex-direction: column;

      .amount {
        display: flex;
        justify-content: flex-end;
        font-size: $font-size-xlg;
      }
    }
  }

  .product {
    border-top: 1px solid black;
    display: flex;
    padding: 20px;
    &.delete-intention {
      .info,
      .info-extra .amount,
      .image {
        opacity: 0.4;
      }

      .info,
      .info-extra {
        text-decoration: line-through;
      }
    }

    &:hover {
      background: black;
      cursor: pointer;

      .close {
        fill: white;
      }
    }

    .image {
      display: flex;
      align-items: center;
      margin-right: 10px;

      &,
      img {
        width: 50px;
      }
    }

    .close {
      display: flex;
      justify-content: flex-end;
    }

    .close-icon {
      width: 14px;
      height: 14px;
    }
  }

  .bag {
    position: relative;

    .counter {
      position: absolute;
      right: 0;
      bottom: 0;
      background: $yellow;
      border-radius: 100px;
      color: black;
      text-align: center;
      font-size: 14px;
      padding: 0.2em 0.6em;
    }

    .bag-icon {
      width: 48px;
      fill: white;
    }
  }

  .cart {
    position: fixed;
    background: $cart-background;
    top: 0;
    bottom: 0;
    right: 0;
    width: 600px;
    transition: all .2s ease-in-out;
    padding: $spacing-sm $spacing-md $spacing-xl;
    z-index: 4000;
    overflow-y: scroll;

    > .close {
      display: flex;
      flex-grow: 1;
      justify-content: flex-end;
      fill: white;
      .close-icon {
        width: 24px;
        height: 24px;
        &:hover {
          cursor: pointer;
        }
      }
    }

    > .title {
      display: flex;
      justify-content: center;
      align-items: center;
      margin-bottom: 40px;

      .content {
        color: white;
        text-transform: uppercase;
        font-size: $font-size-xlg;
        margin-left: $spacing-lg;
      }
    }
  }
</style>
