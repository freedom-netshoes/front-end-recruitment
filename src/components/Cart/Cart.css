.bag{
    text-align: center;
    padding: 50px 0px;
}
.bag .icon_title{
    padding: 0px 0px 50px;
}
.bag .icon_bag{
    position: relative;
}
.bag .icon_bag .qty{
    border-radius: 100%;
    background: #dfbd00;
    font-size: .85em;
    position: absolute;
    color: #000;
    padding: 0.8px 6px;
    bottom: -2px;
    right: -3px;
}
.bag .txt{
    text-transform: uppercase;
    color: #fff;
    font-weight: 700;
    font-size: 1.7em;
    margin-left: 20px;
}


.item_bag{
    padding: 25px 0;
    margin: 0px 25px;
    transition: .3s all;
    position: relative;
}
.item_bag:before{
    transition: .3s all;
    display: block;
    content: " ";
    position: absolute;
    background: transparent;
    width: calc(100% + 50px);
    height: 100%;
    top: 0;
    z-index: -1;
    left: -25px;
    cursor: pointer;
}


.item_bag:hover:before{
    display: block;
    content: " ";
    position: absolute;
    background: #131316;
    width: calc(100% + 50px);
    height: 100%;
    top: 0;
    left: -25px;
    cursor: pointer;
}
.item_bag:first-child:after{
    border-top: 2px solid #000;
}
.item_bag:after{
    border-bottom: 2px solid #000;
    display: block;
    content: " ";
    position: absolute;
    width: 100%;
    height: 100%;
    top: 0;
    z-index: -1;
    left: 0px;
    cursor: pointer;
    transition: .3s all;
}
.item_bag:hover:after{
    border-top: 2px solid #000;
    width: calc(100% + 50px);
    left: -25px;
}

.item_bag > div{
    display: inline-block;
    vertical-align: middle;
}
.item_bag .i{
    margin-right: 20px;
}
.item_bag .p{
    margin-right: 50px;
    width: 53%;
    text-align: left;
}
.item_bag .p *:not(.title){
    font-weight: 400;
    color: #999;
    font-size: .85em;
}

.item_bag.remove-product *{
    text-decoration: line-through;
}
.item_bag.remove-product .p,.item_bag.remove-product .x .value,.item_bag.remove-product .i{
    opacity: .2;
}

.item_bag .p .title{
    font-weight: 600;
    color: #fff;
}
.item_bag .x{
    vertical-align: middle;
    text-align: -webkit-right;
}
.item_bag .x .btn{
    width: 14px;
    height: 14px;
    display: block;
    margin-bottom: 17px;
    background: url(./assets/img/btn_close.png);
}
.item_bag .x:hover ~ div{
    display: none;
}
.item_bag .x .btn:hover{
    cursor: pointer;
    background: url(./assets/img/btn_close_white.png);
}

.item_bag .x .value{
    color: #dfbd00;
}


.subtotal > div{
    display: inline-block;
    vertical-align: middle;
    margin: 0px 25px;
    padding: 25px 0;
    color: rgba(255, 255, 255, .2);
    font-size: 1em;
}

.subtotal .txt{
    text-transform: uppercase;
    
}
.subtotal .price{
    vertical-align: middle;
    text-align: -webkit-right;
    min-width: 64%;
}

.subtotal .price .value{
    color: #dfbd00;
    font-size: 1.7em;
}

.subtotal .price .parceling{
    font-size: .8em;
}

.btn_buy{
    margin: 0 25px;
    text-align: center;
}
.btn_buy button{
    width: 100%;
    background: #000000;
    color: #fff;
    border: 0;
    text-transform: uppercase;
    padding: 20px;
    font-weight: 700;
    font-size: 1em;
    font-family: 'Open Sans',Arial, Helvetica, sans-serif;
    cursor: pointer;
}