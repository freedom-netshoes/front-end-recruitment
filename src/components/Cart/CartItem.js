import React from 'react';
import PropTypes from 'prop-types';
import ImgItemDefault from './assets/img/img_bag_1.jpg'

const hoverClose = (e) => {
    e.target.parentNode.parentNode.classList.add('remove-product')
}

const houtClose = (e) => {
    e.target.parentNode.parentNode.classList.remove('remove-product')
}


const CartItem = ({ title, price, style, availableSizes, currencyFormat, installments, returnPriceWithInstallments,  onClick }) => {
    return (
        <div className="item_bag">
            <div className="i">
                <img src={ImgItemDefault} alt={title} />
            </div>
            <div className="p">
                <div className="title">{title}</div>
                
                <span className="size">{availableSizes.map((availableSize, index, array) => (array.length - 1 === index ? `${availableSize} ` : `${availableSize}, `))}|</span>
                <span className="color">{style}</span>
                <div className="qty">Quantidade:
                    <span className="value">1</span>
                </div>
            </div>
            <div className="x">
                <span onClick={onClick} onMouseOver={hoverClose} onMouseOut={houtClose} className="btn"></span>
                <span className="value">{currencyFormat} <b>{price}</b></span>
            </div>
        </div>
    );
}

CartItem.propTypes = {
    title: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    currencyFormat: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired
}

export default CartItem;
