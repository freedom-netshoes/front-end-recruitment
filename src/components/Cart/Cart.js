import React from 'react';
import PropTypes from 'prop-types';
import BagImage from './assets/img/icon_bag.png'
import CartItem from './CartItem';

const splitPrice = (valor) => {
    var inteiro = valor.toFixed(2).split('.')[0];
    var decimal = valor.toFixed(2).split('.')[1];
    return `${inteiro},${decimal}`;
}


const returnPriceWithInstallments = (price, installments) => {
    var num = price / installments
    if (installments) {
        var n = num.toFixed(2);
        return n
    } else {
        return price
    }

}

const Cart = ({ items, total, currencyFormat, removeFromCart, addQuantity }) => {
    return (
        
        <div className="bag">
            <div className="icon_title">
                <span className="icon_bag">
                    <img src={BagImage} alt="Sacola" />
                    <span className="qty">{items.length}</span>
                </span>
                <span className="txt">Sacola</span>
            </div>
            <div className="cart_items">
                {
                    items.map(item => (<CartItem key={item.id} {...item} onClick={() => removeFromCart(item.id)} />))
                }
            </div>

            <div className="subtotal">
                <div className="txt">Subtotal</div>
                <div className="price">
                    <div className="value">{currencyFormat} <b>{splitPrice(total)}</b></div>
                    <div className="parceling">ou em at√© 10 X R$ {returnPriceWithInstallments(total,10)}</div>
                </div>
            </div>
            <div className="btn_buy">
                <button>comprar</button>
            </div>
        </div>
    );
}

Cart.propTypes = {
    items: PropTypes.array,
    total: PropTypes.number,
    currencyFormat: PropTypes.string,
    removeFromCart: PropTypes.func.isRequired
}

export default Cart;
