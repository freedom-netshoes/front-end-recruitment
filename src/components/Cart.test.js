import { shallow } from 'vue-test-utils'
import Cart from './Cart'

describe('Cart.test.js', () => {
  let cmp

  beforeEach(() => {
    cmp = shallow(Cart, {
      propsData: {
        open: true,
        cart: []
      }
    })
  })

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot()
  })
})