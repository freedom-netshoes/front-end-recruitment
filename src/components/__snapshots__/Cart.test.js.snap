// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart.test.js has the expected html structure 1`] = `
<div
  class="cart"
  style="right: 0px;"
>
  <div
    class="close"
  >
    <!---->
  </div>
   
  <div
    class="title"
  >
    <div
      class="bag"
    >
      <!---->
       
      <div
        class="counter"
      >
        
        0
      
      </div>
    </div>
     
    <div
      class="content"
    >
      
      sacola
    
    </div>
  </div>
   
  <div
    class="list"
  />
   
  <div
    class="subtotal"
  >
    
    subtotal
    
    <div
      class="amount-container"
    >
      <div
        class="amount"
      >
        <div
          class="currency"
        >
          
          R$
        
        </div>
         
        <div
          class="price"
        >
          
          0
        
        </div>
         
        <div
          class="price-decimal"
        >
          
          ,00
        
        </div>
      </div>
       
      <!---->
    </div>
  </div>
   
  <button
    class="button big"
  >
    
    comprar
  
  </button>
</div>
`;
